NumericOutcome.md

```diff
@@ -0,0 +1,542 @@
+be thought of as a general candidate for this particular consideration.
+
+To help with intuition on this matter, consider an arbitrary range of three digit numbers in base 10.
+To capture the same range in base 2 we need 10 digit binary numbers.
+However, a random three digit number in base 10 is expected to have a digit sum of 15, while a random ten digit binary number expects a digit sum of only 5!
+Thus we should expect base 2 to outperform base 10 by around 3x on average.
+This is because using binary results in a compression where each row in the diagram above has only a single element, which corresponds
+to binary compression's ability to efficiently reach the largest possible number of digits ignored which itself covers the largest number of cases.
+Meanwhile in a base like 10, each row can take up to 9 CETs before moving to a larger number of digits ignored (and cases covered).
+Another way to put this is that the inefficiency of base 10 which seems intuitive at small scales is actually equally present at *all scales*!
+
+One final abstract way of intuiting that base 2 is optimal is the following:
+We wish to maximize the amount of information that we may ignore when constructing CETs, because abstractly every bit of information ignored
+in a CET doubles the number of cases covered with a single transaction and signature.
+Thus, if we use any base other than 2, say 10, then we will almost always run into situations where redundant information is needed because we can
+only ignore a decimal digit at a time where a decimal digit has 3.3 bits of information.
+Meanwhile in binary where every digit encodes exactly a single bit of information, we are able to perfectly ignore all redundant bits of information
+resulting in some number near 3.3 times fewer CETs on average.
+
+#### Optimizations
+
+Because `start` and `end` are outliers to the general grouping pattern, there are optimizations that could potentially be made when they are added.
+
+Consider the example in base 10 of the range `[2200, 4999]` which has the endpoints `2200` and `4999` along with the groupings
+
+```
+2201, 2202, 2203, 2204, 2205, 2206, 2207, 2208, 2209,
+221_, 222_, 223_, 224_, 225_, 226_, 227_, 228_, 229_,
+23__, 24__, 25__, 26__, 27__, 28__, 29__,
+
+3___,
+
+40__, 41__, 42__, 43__, 44__, 45__, 46__, 47__, 48__,
+490_, 491_, 492_, 493_, 494_, 495_, 496_, 497_, 498_,
+4990, 4991, 4992, 4993, 4994, 4995, 4996, 4997, 4998
+```
+
+This grouping pattern captures the exclusive range `(2200, 4999)` and then adds the endpoints in ad-hoc to get the inclusive range `[2200, 4999]`.
+But this method misses out on a good amount of compression as re-introducing the endpoints allows us to replace the first two rows with
+a single `22__` and the last 3 rows with just `4___`.
+
+More generally, if the unique digits of `start = (x1)(x2)...(xn)` then if `x(n-i+1) = x(n-i+2) = ... = xn = 0`, then the first `i` rows
```

Comment:
I don't get this sentence. Actually tbh I didn't get this optimization until looking at the code example.
Maybe:
More generally, let the unique digits of `start` be `start = (x1)(x2)...(xn)`. Then if `x(n-i+1) = x(n-i+2) = ... = xn = 0`, then the first `i` rows...

Also I think it could be useful to add an example for this (or just extend the previous one)

Reply:
Just noticed you actually have the example just above... But the sentence could still use some rewording IMHO.